hack/lib/golang.sh

```diff
@@ -972,3 +972,14 @@ kube::golang::build_binaries() {
     done
   fi
 }
+
+# Returns a go package pattern to match all packages in the workspace
+#
+# Use like:
+# kube::util::read-array targets < <(kube::golang::workspace_all)
+#
+# Doing it this way is MUCH faster than simply saying "all", and there doesn't
+# seem to be a simpler way to express "this whole workspace".
+kube::golang::workspace_all() {
+  go list -f '{{.Path}}/...' -m
```

Comment:
This emits a slightly different output that should be compatible but ... need to make sure none of our scripts trip over this.

```console
$ go list -f '{{.Path}}/...' -m
k8s.io/kubernetes/...
k8s.io/api/...
k8s.io/apiextensions-apiserver/...
k8s.io/apimachinery/...
k8s.io/apiserver/...
k8s.io/cli-runtime/...
k8s.io/client-go/...
k8s.io/cloud-provider/...
k8s.io/cluster-bootstrap/...
k8s.io/code-generator/...
k8s.io/component-base/...
k8s.io/component-helpers/...
k8s.io/controller-manager/...
k8s.io/cri-api/...
k8s.io/cri-client/...
k8s.io/csi-translation-lib/...
k8s.io/dynamic-resource-allocation/...
k8s.io/endpointslice/...
k8s.io/kms/...
k8s.io/kube-aggregator/...
k8s.io/kube-controller-manager/...
k8s.io/kube-proxy/...
k8s.io/kube-scheduler/...
k8s.io/kubectl/...
k8s.io/kubelet/...
k8s.io/metrics/...
k8s.io/mount-utils/...
k8s.io/pod-security-admission/...
k8s.io/sample-apiserver/...
k8s.io/sample-cli-plugin/...
k8s.io/sample-controller/...
```

versus:
```console
$ go work edit -json | jq -r '.Use[].DiskPath + "/..."'
./...
./staging/src/k8s.io/api/...
./staging/src/k8s.io/apiextensions-apiserver/...
./staging/src/k8s.io/apimachinery/...
./staging/src/k8s.io/apiserver/...
./staging/src/k8s.io/cli-runtime/...
./staging/src/k8s.io/client-go/...
./staging/src/k8s.io/cloud-provider/...
./staging/src/k8s.io/cluster-bootstrap/...
./staging/src/k8s.io/code-generator/...
./staging/src/k8s.io/component-base/...
./staging/src/k8s.io/component-helpers/...
./staging/src/k8s.io/controller-manager/...
./staging/src/k8s.io/cri-api/...
./staging/src/k8s.io/cri-client/...
./staging/src/k8s.io/csi-translation-lib/...
./staging/src/k8s.io/dynamic-resource-allocation/...
./staging/src/k8s.io/endpointslice/...
./staging/src/k8s.io/kms/...
./staging/src/k8s.io/kube-aggregator/...
./staging/src/k8s.io/kube-controller-manager/...
./staging/src/k8s.io/kube-proxy/...
./staging/src/k8s.io/kube-scheduler/...
./staging/src/k8s.io/kubectl/...
./staging/src/k8s.io/kubelet/...
./staging/src/k8s.io/metrics/...
./staging/src/k8s.io/mount-utils/...
./staging/src/k8s.io/pod-security-admission/...
./staging/src/k8s.io/sample-apiserver/...
./staging/src/k8s.io/sample-cli-plugin/...
./staging/src/k8s.io/sample-controller/...
```

Reply:
Yeah, that doesn't work, but the absolute paths in .Dir do, and we're using that approach elsewhere already.